code/modules/atmospherics/machinery/components/components_base.dm

```diff
@@ -65,9 +65,10 @@
 // Pipenet stuff; housekeeping
 
 /obj/machinery/atmospherics/components/nullifyNode(i)
-	// Every node has a parent pipeline and an air associated with it.
-	nullifyPipenet(parents[i])
-	QDEL_NULL(airs[i])
+	if(nodes[i])
+		if(parents[i] != null)
+			nullifyPipenet(parents[i])
+			QDEL_NULL(airs[i])
```

Comment:
```suggestion
	// Every node has a parent pipeline and an air associated with it.
	if(parents[i])
		nullifyPipenet(parents[i])
	if(airs[i])
		QDEL_NULL(airs[i])
```
Read and *understand* #3089, please.

In short, `if(nodes[i])` shouldn't even be a thing since it wrongly assumes that unconnected nodes don't have air nor pipenet. But, actually, unconnected nodes do have both normally.

Reply:
Nope, explosions could delete it and fuck it up